src/Features/CSharp/Portable/Completion/KeywordRecommenders/BoolKeywordRecommender.cs

```diff
@@ -22,31 +22,32 @@ protected override bool IsValidContextWorker(int position, CSharpSyntaxContext c
     {
         var syntaxTree = context.SyntaxTree;
         return
+            (context.IsGenericTypeArgumentContext && !context.TargetToken.GetRequiredParent().HasAncestor<XmlCrefAttributeSyntax>()) ||
             context.IsAnyExpressionContext ||
+            context.IsCrefContext ||
```

Comment:
sorted these for consistency in a lot of our recommender types.  Added `IsExtensionForTypeContext` check in all of them.